docs/specifications/tests/DNSSEC-TP/dnssec15.md

```diff
@@ -0,0 +1,176 @@
+# DNSSEC15: Existence of CDS and CDNSKEY
+
+## Test case identifier
+**DNSSEC15**
+
+## Objective
+
+CDS and CDNSKEY record types are defined in [RFC 7344] and [RFC 8878].
+Both record types are optional in a zone. The objective of this test
+case is to verify that they are correctly set-up, if included in the
+zone.
+
+If an CDS record is included in the zone, the corresponding CDNSKEY
+record should also be included ([RFC 7344][RFC 7344, section 4],
+section 4).
+
+The CNS and CDNSKEY RRsets should be consistent between all name
+servers for the zone in question.
+
+If there are both CDS RRs and CDNSKEY RRs in the zone they must match in 
+content ([RFC 7344][RFC 7344, section 4], section 4). It means that both
+must be derived from the same DNSKEY or both being "delete" CDS and
+CDNSKEY.
+
+If a name server has issues covered by [Basic04] (basic name server
+issues) no messages will be outputted from this test case.
+
+## Inputs
+
+* "Child Zone" - The domain name to be tested.
+
+## Summary
+
+* [INFO] message if CDS RRset is found on any name server for the zone.
+* [INFO] message if CDNSKEY RRset is found on any name server for the zone.
+* [NOTICE] message if only one type is found.
+* [ERROR] message if the two RRsets (CDS and CDNSKEY) both exist but do not 
+  match.
+* [ERROR] message if not all servers have the same CDS RRset (if CDS exists)
+* [ERROR] message if not all servers have the same CDNSKEY RRset (if CDNSKEY
```

Comment:
depends on the interpretation of the RFC and my previous question

Reply:
I'm not sure what you mean. Can you specify?